cryptoconditions/condition.py

```diff
@@ -232,87 +314,86 @@ def serialize_binary(self):
+        payload = {'fingerprint': self.hash, 'cost': self.cost}
+        if condition_class.TYPE_CATEGORY == 'compound':
+            subtype_ids = [
+                TypeRegistry.find_by_name(subtype)['type_id']
+                for subtype in self.subtypes
+            ]
+            bits = ['0' for bit in range(5)]
+            for subtype_id in subtype_ids:
+                bits[subtype_id] = '1'
+            bitstring = ''.join(bits).rstrip('0')
+            payload['subtypes'] = bitstring
+        return {condition_class.TYPE_ASN1: payload}
+
+    # TODO ILP Clarification NEEDED
+    # The asn1 json payload that ILP uses in the JS implementation differs from
+    # the dictionary payload generated by pyasn1 -- hence the question:
+    # Is there a standard JSON representation of a condition?
+    def to_asn1_json(self):
+        condition_type = TypeRegistry.find_by_type_id(self.type_id)
+        condition_class = condition_type['class']
+        asn1_dict = {
+            'type': condition_class.TYPE_ASN1_CONDITION,
+            'value': {
+                'fingerprint': self.hash,
+                'cost': self.cost,
+            },
         }
 
-    def parse_dict(self, data):
-        """
-
-        Args:
-            data (dict):
-        Returns:
-            Condition with payload
-        """
-        self.type_id = data['type_id']
-        self.bitmask = data['bitmask']
-
-        self.hash = base58.b58decode(data['hash'])
-        self.max_fulfillment_length = data['max_fulfillment_length']
-
+        if condition_class.TYPE_CATEGORY == 'compound':
+            subtype_ids = [
+                TypeRegistry.find_by_name(subtype)['type_id']
+                for subtype in self.subtypes
+            ]
+            bits = ['0' for bit in range(5)]
+            for subtype_id in subtype_ids:
+                bits[subtype_id] = '1'
+            bitstring = ''.join(bits).rstrip('0')
+            asn1_dict['value']['subtypes'] = bitstring
+        return asn1_dict
+
+    # TODO ILP Clarification NEEDED
     def validate(self):
         """
         Ensure the condition is valid according the local rules.
 
-        Checks the condition against the local bitmask (supported condition types)
-        and the local maximum fulfillment size.
+        Checks the condition against the local subtypes (supported
+        condition types) and the local maximum fulfillment size.
 
         Returns:
             bool: Whether the condition is valid according to local rules.
         """
         # Get class for type ID, throws on error
-        TypeRegistry.get_class_from_type_id(self.type_id)
+        TypeRegistry.find_by_type_id(self.type_id)
 
-        # Bitmask can have at most 32 bits with current implementation
-        if self.bitmask > Condition.MAX_SAFE_BITMASK:
-            raise ValueError('Bitmask too large to be safely represented')
+        # Subtypes can have at most 32 bits with current implementation
```

Comment:
Yeah, it is somewhat clumsy. This was done quickly, as the JS implementation had yet to be updated there. See https://github.com/interledgerjs/five-bells-condition/issues/67